spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/HystrixGatewayFilterFactory.java

```diff
@@ -272,6 +273,10 @@ public Config setSetter(Setter setter) {
 			ServerHttpRequest request = this.exchange.getRequest().mutate()
 					.uri(requestUrl).build();
 			ServerWebExchange mutated = exchange.mutate().request(request).build();
+			// Before we continue on remove the already routed attribute since the
+			// fallback may go back through the route handler if the fallback
+			// is to another route in the Gateway
+			mutated.getAttributes().remove(GATEWAY_ALREADY_ROUTED_ATTR);
```

Comment:
Retry filter does this too: https://github.com/spring-cloud/spring-cloud-gateway/blob/259d4bd8add36bc5a8e5f3891d5e0492fcfb1270/spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/RetryGatewayFilterFactory.java#L204-L211

I wonder if at some point we should consolidate. Not needed for this PR.

Reply:
Yeah we already have a utility method to add the attribute, are you thinking we have one to remove it as well?  It wouldnt be much to add it in this PR.  I can do that tomorrow.

Reply:
Yeah, something like that

Reply:
@spencergibb method added

Reply:
:+1: